### ISSUE TITLE
request: way to manually tag certain sets to be omitted from calculation

### ISSUE THREAD

**Issue by VestboyMyst** [ripgarpr/garpr#73]
this is needed in several instances:

- when a player is DQed/has to leave early after playing >=1 set in a bracket
- when TOs mark DQs as 0-2 losses
- when TOs do 'DIY seeding' for topX brackets involving giving some players random losses to put them into losers

smashgg allows you to see if a set was won by DQ so those can probably be added to the omit list automatically. admins will need a manual method for sure, but there should be some checks planned to prevent abuse

**Comment by jhertz** [comment]
Can assign this to me, ive been planning to add this anyway

**Comment by BrandonCookeDev** [comment]
Update:
@VestboyMyst @jhertz 
Added check into smash gg to omit DQ matches.

**Comment by VestboyMyst** [comment]
so im imagining something like this:
when you go to a bracket page, there are a series of checkboxes next to each match. if you check it off to omit, a field appears where you can enter a small note which is saved and viewed by the public, so we can say 'omitted because X was drunk and playing an alt' or something

**Comment by jschnei** [comment]
is this currently the highest-priority enhancement? @VestboyMyst

**Comment by jessemtso** [comment]
ima do this one

**Comment by BrandonCookeDev** [comment]
What's the progress on this @jessemtso ? 
I have lots of people bitching at me because our TO's refuse to mark DQ's as -1 in challonge.

**Comment by jessemtso** [comment]
i havent had time to do this. feel free to have at it, otherwise i will when i have some time

**Comment by BrandonCookeDev** [comment]
Is this something like a checkmark on a submitted tournament that disqualifies a set from ranking?

**Comment by jschnei** [comment]
it's essentially this.
you'll probably need to do a db migration, but hopefully this should be easier now.

**Comment by VestboyMyst** [comment]
so we'll have a checkbox next to each line of names on the bracket page? i guess check to omit, unchecked to count. sounds good

**Comment by BrandonCookeDev** [comment]
yeah and it will be good for the UI to show the row as just grey so the users are never confused about what counts and what doesn't

**Comment by BrandonCookeDev** [comment]
So this will be very simple but two new attributes need to be added to every 'match' document embedded in tournament.matches.
We need:
an '_id' field with a UID to identify specific sets 
an 'excluded' field that defines a boolean of if it's excluded.

I was trying to script the addition of these properties but in my inexperience I've  squandered too much time on it.

**Comment by BrandonCookeDev** [comment]
UPDATE: 
Alright, so I'm approaching completion on this fix.
The following script can be used to alter the database for this implementation
``` 
var t = db.tournaments.find();
t.forEach(function(tournament){
   var id = tournament._id;
   var matches = tournament.matches;
   var new_matches = [];

   for(var i = 0; i < matches.length; i++){
       var match = matches[i];
       match['match_id'] = i;
       match['excluded'] = false;
       new_matches.push(match);
   };
   db.tournaments.update({'_id':id}, {$set: { 'matches': new_matches} });
});
```

Once here we should be able to use PyMongo or ORM to alter the match exclusion. 
After this we can edit the UI to show grey rows for any matches that excluded = true.

**Comment by BrandonCookeDev** [comment]
@VestboyMyst 
Issue is solved by #125
